test/tyche/YieldDirector.test.js

```diff
@@ -505,6 +515,21 @@ describe("YieldDirector", async () => {
         await expect(bobBalance).is.equal(donatedAmount.add("2"));
     });
 
+    it("should redeem all tokens on bahalf of others", async () => {
```

Comment:
should we also add a test to make sure it reverts if .givePermissionToRedeem isn't called?

Reply:
done